### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by Placu95**
Add documentation for:

- simulation in indoor environment
- simulation in real map
- simulation with GPS trace

### CODE REVIEW

File: alchemist/src/orchid/resources/wiki/usage/indoor.md [hunk 1]
```diff
@@ -4,4 +4,16 @@ title: "Simulation of indoor environments"
 
 ---
 
+In order to load the map of the indoor environment, you can use the {{ anchor('ImageEnvironment') }}.
 
+ImageEnvironment loads the map as raster image from file, interpreting the black pixels as obstacles
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/indoor.md]: Actually the color can be choosen. Can you update?

File: alchemist/src/orchid/resources/wiki/usage/indoor.md [hunk 2]
```diff
@@ -4,4 +4,16 @@ title: "Simulation of indoor environments"
 
 ---
 
+In order to load the map of the indoor environment, you can use the {{ anchor('ImageEnvironment') }}.
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/indoor.md]: remove "the"

File: alchemist/src/orchid/resources/wiki/usage/indoor.md [hunk 3]
```diff
@@ -4,4 +4,16 @@ title: "Simulation of indoor environments"
...
 
+ImageEnvironment loads the map as raster image from file, interpreting the black pixels as obstacles
+(areas that cannot be accessed by nodes like wall).
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/indoor.md]: "wall-like areas not accessible to nodes"

File: alchemist/src/orchid/resources/wiki/usage/indoor.md [hunk 4]
```diff
@@ -4,4 +4,16 @@ title: "Simulation of indoor environments"
...
+(areas that cannot be accessed by nodes like wall). 
+The color of the pixel that represent obstacle can be set as constructor's parameter of the environment.
+
+In this example the image with the map is in theclasspath in the folder `images`:
+```yaml
+incarnation: protelis
+environment:
+  type: ImageEnvironment
+  parameters: [images/foo.png]
+```
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/indoor.md]: Can you add a few details on network models? Namely: how to configure connections in such a way that they do not get across walls

Also, a snapshot may be of help to better understand how it works.
Ideally, also an example yaml file that makes nodes move with a brownian move in an environment with obstacles

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+---
+
+title: Maps and GPS traces
+
+---
+
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/maps.md]: Introduction: Alchemist can simulate real-world scenarios using map data from openstreetmaps... etc.

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+To set a simulation that uses a map and possibly the GPS tracks,
+three parameters of the configuration file must be appropriately defined:
+0. Environment
+0. Displacement
+0. Program to follow the GPS traces
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/maps.md]: Unclear what this means

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+## Setting environment with map
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/maps.md]: Creating? Loading?

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+0. Program to follow the GPS traces
+
+## Setting up a map environment
+
+In order to run simulations on real world maps, an appropriate environment must be selected.
+
+Currently, the only environment supporting maps is {{ anchor('OSMEnvironment') }}. 
+
+The only mandatory parameter is the path of the file with the map to be loaded.
+For the optional parameters, refer to the documentation of the environment.
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/maps.md]: Aheemmm... nope. That's the documentation. Please expand.

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+you need to use the displacement {{ anchor('FromGPSTrace') }}.
+The displacement `FromGPSTrace` require the following parameters:
+
+0. number of nodes to displace
+0. file with the GPS traces to use
+0. boolean that indicates if the list of GPS trace is cyclic.
+   If `true` and the GPS traces are less than the number of nodes to be displaced,
+   then the traces are cyclically re-used to displace nodes.
+0. a {{ anchor('GPSTimeAlignment') }} to define how to align the time of all the GPS points of all GPS traces. 
+    There are present several strategy in the package {{ anchor('it.unibo.alchemist.boundary.gpsload.api') }}
```
Comment (DanySK) [inline alchemist/src/orchid/resources/wiki/usage/maps.md]: Add a subsection with a description of the available `GPSTimeAlignment`s

**Comment by DanySK** [comment]
@Placu95 please reply to every comment, so that we know what's closed and what's still open

File: alchemist/src/orchid/resources/wiki/usage/indoor.md (hunk 1 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/indoor.md]: resolved

File: alchemist/src/orchid/resources/wiki/usage/indoor.md (hunk 2 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/indoor.md]: resolved

File: alchemist/src/orchid/resources/wiki/usage/indoor.md (hunk 3 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/indoor.md]: resolved

File: alchemist/src/orchid/resources/wiki/usage/indoor.md (hunk 4 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/indoor.md]: resolved

File: alchemist/src/orchid/resources/wiki/usage/maps.md (hunk 5 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: I think it was already done

File: alchemist/src/orchid/resources/wiki/usage/maps.md (hunk 6 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: I remove it

File: alchemist/src/orchid/resources/wiki/usage/maps.md (hunk 7 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: I think it was already done

File: alchemist/src/orchid/resources/wiki/usage/maps.md (hunk 8 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: done

File: alchemist/src/orchid/resources/wiki/usage/maps.md (hunk 9 above)
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: added

File: alchemist/src/orchid/resources/wiki/usage/maps.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+      parameters: [1497, "vcmuser.gpx", false, "AlignToTime", 1365922800, false, false]
+```
+
+## Following GPS traces
+
+In order to obtain a behavior that moves the nodes following the GPS traces, the program can use one of the
+following types of actions:
+
+0. {{ anchor('GPSTraceWalker') }} -> to follow the GPS trace, interpolating its points my moving along roads
+0. {{ anchor('ReproduceGPSTrace') }} -> to follow the GPS trace as they are
```
[the remark this answers is not in the document]
Comment (Placu95) [reply alchemist/src/orchid/resources/wiki/usage/maps.md]: done

**Pull request merged**